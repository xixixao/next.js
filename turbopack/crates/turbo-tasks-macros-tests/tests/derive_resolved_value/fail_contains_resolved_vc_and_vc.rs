#![allow(dead_code)]

use turbo_tasks::{ResolvedValue, ResolvedVc, Vc};

#[derive(ResolvedValue)]
struct ContainsResolvedVcAndVc {
    a: ResolvedVc<i32>,
    b: Vc<i32>,
}

fn main() {}
