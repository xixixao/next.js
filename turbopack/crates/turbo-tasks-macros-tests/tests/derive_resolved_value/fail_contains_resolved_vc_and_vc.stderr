error[E0277]: the trait bound `Vc<i32>: ResolvedValue` is not satisfied
 --> tests/derive_resolved_value/fail_contains_resolved_vc_and_vc.rs:8:8
  |
8 |     b: Vc<i32>,
  |        ^^^^^^^ the trait `ResolvedValue` is not implemented for `Vc<i32>`
  |
  = help: the following other types implement trait `ResolvedValue`:
            &T
            &mut T
            ()
            (A, Z, Y, X, W, V, U, T)
            (B, A, Z, Y, X, W, V, U, T)
            (C, B, A, Z, Y, X, W, V, U, T)
            (D, C, B, A, Z, Y, X, W, V, U, T)
            (E, D, C, B, A, Z, Y, X, W, V, U, T)
          and $N others
note: required by a bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
 --> tests/derive_resolved_value/fail_contains_resolved_vc_and_vc.rs:5:10
  |
5 | #[derive(ResolvedValue)]
  |          ^^^^^^^^^^^^^ required by this bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
  = note: this error originates in the derive macro `ResolvedValue` (in Nightly builds, run with -Z macro-backtrace for more info)
